CRM/ManualDirectDebit/Upgrader.php

```diff
@@ -7,11 +8,124 @@
+    $this->uninstallCustomInformation();
+  }
+
+  /**
+   *  Uninstall custom information
+   */
+  private function uninstallCustomInformation() {
+    $customValuesForUninstall = [
+      [
+        "entityType" => "OptionGroup",
+        "searchValue" => "direct_debit_codes",
+      ],
+      [
+        "entityType" => "OptionGroup",
+        "searchValue" => "direct_debit_originator_number",
+      ],
+      [
+        "entityType" => "CustomGroup",
+        "searchValue" => "direct_debit_mandate",
+      ],
+      [
+        "entityType" => "CustomGroup",
+        "searchValue" => "direct_debit_information",
+      ],
+      [
+        "entityType" => "UFGroup",
+        "searchValue" => "Direct Debit Information",
+        "searchField" => "title",
+      ],
+      [
+        "entityType" => "PaymentProcessor",
+        "searchValue" => "OfflineDirectDebit",
+        "searchField" => "payment_processor_type_id",
+      ],
+      [
+        "entityType" => "PaymentProcessorType",
+        "searchValue" => "OfflineDirectDebit",
+      ],
+      [
+        "entityType" => "ContributionPage",
+        "searchValue" => "Renew offline auto-renewal memberships",
+        "searchField" => "title",
+      ],
+    ];
+
+    foreach ($customValuesForUninstall as $customValue) {
+      $this->deleteEntityRecord(
+        $customValue['entityType'],
+        $customValue['searchValue'],
+        isset($customValue['searchField']) ? $customValue['searchField'] : ""
+      );
+    }
+  }
+
+  /**
+   * Deletes a record for the specified CiviCRM entity based on
+   * the search field and search value.
+   *
+   * @param string $entityType
+   *   The entity type for the record we want to remove (e.g OptionValue,
+   *   PaymentProcessor ..etc). it should be a valid Custom or CiviCRM core
+   *   entity type.
+   * @param string $searchValue
+   *   The search value to find the record that we want to remove
+   * @param string $searchField
+   *   The field that we search the search value against.
+   */
+  private function deleteEntityRecord($entityType, $searchValue, $searchField = 'name') {
+    civicrm_api3($entityType, 'get', [
+      $searchField => $searchValue,
+      'api.' . $entityType . '.delete' => ['id' => '$value.id'],
+    ]);
+  }
+
+  /**
+   * Will install the DirectDebit payment processor
+   */
+  private function createDirectDebitPaymentProcessorType() {
+    $paymentProcessorType = [
+      "name" => "OfflineDirectDebit",
+      "title" => ts("Offline Direct Debit"),
+      "description" => ts("Payment processor"),
+      "is_active" => 1,
+      "user_name_label" => ts("Offline Direct Debit"),
+      "class_name" => "Payment_Manual",
+      "billing_mode" => "1",
+      "is_recur" => "1",
+      "payment_type" => CRM_Core_Payment::PAYMENT_TYPE_DIRECT_DEBIT,
+    ];
+    civicrm_api3('PaymentProcessorType', 'create', $paymentProcessorType);
+  }
+
+  /**
+   * Will install the DirectDebit payment processor
```

Comment:
Please rename the method description to : 

Installs the 'Direct Debit' payment processor